tutorials/plugins/gdnative/gdnative-cpp-scons.rst

```diff
@@ -0,0 +1,872 @@
+
+If we combine the library filename with the platform/bits-specific directory path, then we can determine the full appropriate filename to use for the dynamic library's output location.
+
+.. code-block:: python
+
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+    lib_name = ARGUMENTS.get("name", "libdefault")
+    lib_path = ARGUMENTS.get("lib", "bin/")
+
+    # Force expected formatting.
+    # Ensures we view Windows-style and filepath-like 
+    # directory paths as typical UNIX directory paths.
+    lib_path = lib_path.split("\\").join("/")
+    if lib_path[-1:] != "/":
+        lib_path += "/"
+
+    platform_dir = ""
+
+    if platform == "osx":
+        platform_dir = "osx" + str(bits) + "/"
+    elif platform == "linux":
+        platform_dir = "x11" + str(bits) + "/"
+    elif platform == "windows":
+        platform_dir = "win" + str(bits) + "/"
+
+    # No need to include the file extension.
+    # scons will determine it automatically based on parameters.
+    library_file = lib_name + "." + platform + "." + str(bits)
+    final_lib_path = lib_path + platform_dir + library_file
+
+Each operating system will also have different flags to pass to the C++ compiler and linker we use.
+
+All of the available options for each platform are really beyond the scope of this tutorial, but you can find more information by examining the compiler options for each individual compiler. In this example, the compiler options specified are related to ``g++`` for OSX/Linux and ``cl`` for Windows.
+
+An overview of the compiler and linker options for ``g++`` can be found here: https://developers.redhat.com/blog/2018/03/21/compiler-and-linker-flags-gcc/
+
+An overview of the compiler options for ``cl`` can be found here: https://msdn.microsoft.com/en-us/library/19z1t1wy.aspx
+
+.. code-block:: python
+
+    target = ARGUMENTS.get("target", "debug")
+    platform = ARGUMENTS.get("platform", "windows")
+    
+    if platform == "osx":
+        env.Append(CCFLAGS = ['-g','-O3', '-arch', 'x86_64'])
+        env.Append(LINKFLAGS = ['-arch', 'x86_64'])
+    
+    elif platform == "linux":
+        env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++14'])
+    
+    elif platform == "windows":
+        if target == "debug":
+            env.Append(CCFLAGS = ['-EHsc', '-D_DEBUG', '-MDd'])
+        else:
+            env.Append(CCFLAGS = ['-O2', '-EHsc', '-DNDEBUG', '-MD'])
+
+Notice that we define the compilation and linking flags by assigning the command line options as an array of strings to special property names. These names (like ``CCFLAGS``) are specifically used by scons to build whatever we want from the ``Environment`` we've setup in the script.
+
+Phew! Okay, so that's the name of our library, where the outputted libraries will end up, and the flags for each OS.
```

Comment:
Where the *built* libraries will end up? This reads a bit awkward to me.

Compiled? Assembled?